src/main/java/org/folio/circulation/services/ItemForTlrService.java

```diff
@@ -69,7 +70,27 @@ private Result<List<Item>> refusePageRequestWhenNoAvailablePageableItemsExist(Re
       return failedValidation(message, INSTANCE_ID, request.getInstanceId());
     }
 
-    return of(() -> availablePageableItems);
+    List<Item> finalAvailablePageableItems;
+    if (request.geItemLocationCode() != null) {
+      finalAvailablePageableItems = availablePageableItems.stream()
+        .filter(item -> request.geItemLocationCode().equals(item.getLocation().getCode()) ||
+            request.geItemLocationCode().equals(item.getLocation().getLibrary().getCode()) ||
+            request.geItemLocationCode().equals(item.getLocation().getCampus().getCode()) ||
+            request.geItemLocationCode().equals(item.getLocation().getInstitution().getCode())
```

Comment:
Can we extract this into a method in class `Item`? Perhaps `isAtLocation(String locationCode)`? And please beware of possible NPEs.

Reply:
Moved to method 
library, campus and institution have @NonNull annotation in Location object